app/frontend/src/subjectsFilter.js

```diff
@@ -0,0 +1,62 @@
+document.addEventListener('DOMContentLoaded', function() {
+  var subjectFilter, filteredSubjectNames;
+  var checkedSubjectElements = [];
+  var duplicateSubjectElements = [];
+
+  const subjectSearchInput = document.getElementById('job-specification-form-subject-search');
+  const subjectElements = document.querySelector('.checkboxes-with-scroll');
+
+  const subjectArray = [...subjectElements.children];
+  subjectArray.shift();
+  const subjectNames = subjectArray.map(ele => ele.children[1].textContent.toLowerCase());
+
+  const subjectCheckboxes = document.querySelectorAll('.subjects-options > .govuk-checkboxes__item > input');
+
+  function displayElement(element) {
+    var duplicateCheckedElement = duplicateSubjectElements.find(function(ele) {
+      return ele.children[1].textContent == element.children[1].textContent;
+    });
+    if ((filteredSubjectNames.includes(element.children[1].textContent.toLowerCase()) || element.children[0].checked) && !duplicateCheckedElement) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  document.querySelector('.js-checkbox').classList.remove('display-none');
+  document.querySelector('.subjects-options').classList.remove('js-disabled-border');
+
+  [...subjectCheckboxes].forEach(function(checkbox) {
+    checkbox.addEventListener('change', function() {
+      if (this.checked) {
+        checkedSubjectElements.push(this.parentElement);
+        var duplicateElement = this.parentElement.cloneNode(true);
+        duplicateSubjectElements.push(duplicateElement);
+        duplicateElement.addEventListener('change', function() {
+          if (!this.checked) {
+            var originalElement = checkedSubjectElements.find(function(ele) {
+              return ele.children[1].textContent == duplicateElement.children[1].textContent;
+            });
+            originalElement.style.display = 'block';
+            this.remove();
+          }
+        });
+        subjectElements.prepend(duplicateElement);
+        this.checked = false;
+        this.parentElement.style.display = 'none';
+      }
+    });
+  });
+
+  subjectSearchInput.addEventListener('input', function(v) {
+    subjectFilter = subjectSearchInput.value.toLowerCase();
+    filteredSubjectNames = subjectNames.filter(function(a) {
+      return a.includes(subjectFilter);
+    });
+
+    for (var i = 0; i < subjectArray.length; i++) {
+      var element = subjectArray[i];
+      displayElement(element) ? element.style.display = 'block' : element.style.display = 'none';
+    }
+  });
+});
```

Comment:
in general I would be tempted to use jquery for DOM selection and because of IE support for some stuff. im 95% that we will carry on using a thing called sizzle which is the DOM selection library used in jquery, so this should mean little to no refactoring needed when jquery is removed